common/defines/NDefines.NAI.lua

```diff
@@ -144,28 +150,46 @@ NDefines.NAI.PEACE_TERMS_FLEET_BASING_BASE_MULT = 0 -- AI desire for fleet basin
-NDefines.NAI.NAVY_BUDGET_FRACTION = 0.4 -- AI will spend a maximum of this fraction of monthly income on navy maintenance (based off wartime costs)
+NDefines.NAI.NAVY_BUDGET_FRACTION = 0.5 -- AI will spend a maximum of this fraction of monthly income on navy maintenance (based off wartime costs)
 	-- NAVY_BUDGET_FRACTION = 0.5, -- AI will spend a maximum of this fraction of monthly income on navy maintenance (based off wartime costs)
-NDefines.NAI.FORT_BUDGET_FRACTION = 0.5 -- AI will spend a maximum of this fraction of monthly income on forts
+NDefines.NAI.FORT_BUDGET_FRACTION = 0.45 -- AI will spend a maximum of this fraction of monthly income on forts
 	-- FORT_BUDGET_FRACTION = 0.4, -- AI will spend a maximum of this fraction of monthly income on forts
+NDefines.NAI.MISSIONARY_MAINTENANCE_SHARE = 0.3	--The most an AI will spend of its income on a missionary
+	--MISSIONARY_MAINTENANCE_SHARE = 0.2 
+--NDefines.NAI.REGIMENTS_PER_GENERAL = 15	--AI will want one general for every this number of regiments (will not exceed free leader pool).
+NDefines.NAI.MIN_SHIPS_FOR_ADMIRAL = 20	--The minimum navy size for the AI to bother with an admiral.
+	--MIN_SHIPS_FOR_ADMIRAL = 10
 
 NDefines.NAI.CANCEL_CONSTRUCTION_SIEGE_PROGRESS = 101 -- buildings don't get destroyed when sieged, not sure why AI cancels constructions at all
 	-- CANCEL_CONSTRUCTION_SIEGE_PROGRESS = 0, -- If chance of fort falling is at least this, AI will cancel constructions in the province
 NDefines.NAI.DESIRED_SURPLUS = 0.10 -- AI will aim for having at least this fraction of their income as surplus when they don't have large savings
 	-- DESIRED_SURPLUS = 0.1, -- AI will aim for having at least this fraction of their income as surplus when they don't have large savings
-NDefines.NAI.BIGSHIP_FRACTION = 0.3	-- The proportion of big ships in an AI navy of light ships and big ships (for coastal sea countries, this fraction is mostly galleys)
+NDefines.NAI.BIGSHIP_FRACTION = 0.35 --The proportion of big ships in an AI navy of light ships and big ships (for coastal sea countries, this fraction is mostly galleys)
 	-- BIGSHIP_FRACTION = 0.4,	-- The proportion of big ships in an AI navy of light ships and big ships (for coastal sea countries, this fraction is mostly galleys)
+NDefines.NAI.TRANSPORT_FRACTION = 0.3 --Max fraction of naval forcelimit that should be transports.
+	--TRANSPORT_FRACTION = 0.5 
 
  -- AI will not develop provinces that have more development than this or DEVELOPMENT_CAP_MULT*original development (whichever is bigger)
 NDefines.NAI.DEVELOPMENT_CAP_BASE = 30 --set to 3 manually if using dev-click setting
 NDefines.NAI.DEVELOPMENT_CAP_MULT = 2  --set to 1 manually if using dev-click setting
 
+NDefines.NAI.ARTILLERY_FRACTION = 0.30 --Ratio of artillery to infantry AI will build.
+	--ARTILLERY_FRACTION = 0.35
+NDefines.NAI.FORCE_COMPOSITION_CHANGE_TECH_LEVEL = 16 --Tech level at which AI will double its artillery fraction.
```

Comment:
Does it make sense to set this to 15 so the AI has some time to build up some artillery before 16?

Reply:
Most of the time AI is sitting not at the force limit, so usually he have room for additional cannons. Ideally AI needs just enough cannons to fill the back row. And it's not like tech 16 is extreme breakpoint, cannons are slowly getting better with tech. 
Also I don't think AI follows this ratio precisely, he probably have some hardcoded eco considerations.